### ISSUE TITLE
"RECS0154 Parameter '' is never used" is a bit annoying on "this" parameters in extension methods

### ISSUE THREAD

**Issue by neoGeneva** [icsharpcode/RefactoringEssentials#199]
This is more of a suggestion than anything else, given the the following code in an ASP.NET MVC project:

``` csharp
public static IHtmlString Div(this HtmlHelper htmlHelper)
{
    var tag = new TagBuilder("div");

    return MvcHtmlString.Create(tag.ToString(TagRenderMode.Normal));
}
```

In this case even though "htmlHelper" isn't used within the method it's still useful from elsewhere as a way to reference the code.

All though I use this pattern quite extensively in my own code base, I'm not 100% sure that I'm right here, but I think removing the check from "this" parameters could be worth considering if others are having htis issue too.

**Comment by Rpinski** [comment]
I wouldn't remove that warning generally, because actually the diagnostic is right here: This method doesn't need to be an extension method at all.

This could be a candidate in case we introduce a configuration mechanism, where you could disable the warning for extension methods especially in your project.

**Comment by neoGeneva** [comment]
I agree, the diagnostic is right, and honestly after seeing the warning raised in my code I'm not sure that what I'm doing is all that clean.

That said, I think that having an unused "this" parameter does feel different from a regular unused parameter, since removing a regular unused parameter just means removing the extra value passed through, while removing the "this" parameter requires the way the method is referenced to be changed in a very different way (e.g. like my original `Html.Div()` would change to `HtmlExtensions.Div()`)

Perhaps it'd be worth giving "unused 'this' parameter" its own error?
